pilot/pkg/security/authn/v1alpha1/applier.go

```diff
@@ -0,0 +1,327 @@
+	authn_filter "istio.io/api/envoy/config/filter/http/authn/v2alpha1"
+	jwtfilter "istio.io/api/envoy/config/filter/http/jwt_auth/v2alpha1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+	"istio.io/istio/pilot/pkg/security/authn"
+	"istio.io/istio/pkg/features/pilot"
+	"istio.io/istio/pkg/log"
+	protovalue "istio.io/istio/pkg/proto"
+)
+
+const (
+	// JwtFilterName is the name for the Jwt filter. This should be the same
+	// as the name defined in
+	// https://github.com/istio/proxy/blob/master/src/envoy/http/jwt_auth/http_filter_factory.cc#L50
+	JwtFilterName = "jwt-auth"
+
+	// AuthnFilterName is the name for the Istio AuthN filter. This should be the same
+	// as the name defined in
+	// https://github.com/istio/proxy/blob/master/src/envoy/http/authn/http_filter_factory.cc#L30
+	AuthnFilterName = "istio_authn"
+
+	// EnvoyTLSInspectorFilterName is the name for Envoy TLS sniffing listener filter.
+	EnvoyTLSInspectorFilterName = "envoy.listener.tls_inspector"
+)
+
+// GetMutualTLS returns pointer to mTLS params if the policy use mTLS for (peer) authentication.
+// (note that mTLS params can still be nil). Otherwise, return (false, nil).
+// Callers should ensure the proxy is of sidecar type.
+func GetMutualTLS(policy *authn_v1alpha1.Policy) *authn_v1alpha1.MutualTls {
+	if policy == nil {
+		return nil
+	}
+	if len(policy.Peers) > 0 {
+		for _, method := range policy.Peers {
+			switch method.GetParams().(type) {
+			case *authn_v1alpha1.PeerAuthenticationMethod_Mtls:
+				if method.GetMtls() == nil {
+					return &authn_v1alpha1.MutualTls{Mode: authn_v1alpha1.MutualTls_STRICT}
+				}
+				return method.GetMtls()
+			default:
+				continue
+			}
+		}
+	}
+	return nil
+}
+
+// collectJwtSpecs returns a list of all JWT specs (pointers) defined the policy. This
+// provides a convenient way to iterate all Jwt specs.
+func collectJwtSpecs(policy *authn_v1alpha1.Policy) []*authn_v1alpha1.Jwt {
+	ret := []*authn_v1alpha1.Jwt{}
+	if policy == nil {
+		return ret
+	}
+	for _, method := range policy.Peers {
+		switch method.GetParams().(type) {
+		case *authn_v1alpha1.PeerAuthenticationMethod_Jwt:
+			ret = append(ret, method.GetJwt())
+		}
+	}
+	for _, method := range policy.Origins {
+		ret = append(ret, method.Jwt)
+	}
+	return ret
+}
+
+// OutputLocationForJwtIssuer returns the header location that should be used to output payload if
+// authentication succeeds.
+func outputLocationForJwtIssuer(issuer string) string {
+	const locationPrefix = "istio-sec-"
+	sum := sha1.Sum([]byte(issuer))
+	return locationPrefix + fmt.Sprintf("%x", sum)
+}
+
+// ConvertPolicyToJwtConfig converts policy into Jwt filter config for envoy.
+func convertPolicyToJwtConfig(policy *authn_v1alpha1.Policy) *jwtfilter.JwtAuthentication {
+	policyJwts := collectJwtSpecs(policy)
+	if len(policyJwts) == 0 {
+		return nil
+	}
+	ret := &jwtfilter.JwtAuthentication{
+		AllowMissingOrFailed: true,
```

Comment:
Suggest defining a const variable (e.g., const DefaultJwtAllowMissingOrFailed = true) to represent the default value of AllowMissingOrFailed, instead of hard-coding the default value in the function.